nf_core/modules/create.py

```diff
@@ -272,27 +283,46 @@ def render_template(self):
             template_stat = os.stat(os.path.join(os.path.dirname(nf_core.__file__), "module-template", template_fn))
             os.chmod(dest_fn, template_stat.st_mode)
 
-    def get_repo_type(self, directory):
+    def get_repo_type(self):
         """
         Determine whether this is a pipeline repository or a clone of
         nf-core/modules
         """
         # Verify that the pipeline dir exists
-        if dir is None or not os.path.exists(directory):
-            raise UserWarning(f"Could not find directory: {directory}")
-
-        readme = os.path.join(directory, "README.md")
-        # Determine repository type
-        if os.path.exists(readme):
-            with open(readme) as fh:
-                if fh.readline().rstrip().startswith("# ![nf-core/modules]"):
-                    return "modules"
-                else:
-                    return "pipeline"
-        else:
+        if dir is None or not os.path.exists(self.directory):
+            raise UserWarning(f"Could not find directory: {self.directory}")
+
+        # Try to find the root directory
+        base_dir = os.path.abspath(self.directory)
+        config_path_yml = os.path.join(base_dir, ".nf-core.yml")
+        config_path_yaml = os.path.join(base_dir, ".nf-core.yaml")
+        while (
+            not os.path.exists(config_path_yml)
+            and not os.path.exists(config_path_yaml)
+            and base_dir != os.path.dirname(base_dir)
+        ):
+            base_dir = os.path.dirname(base_dir)
+            config_path_yml = os.path.join(base_dir, ".nf-core.yml")
+            config_path_yaml = os.path.join(base_dir, ".nf-core.yaml")
+            # Reset self.directory if we found the config file (will be an absolute path)
+            if os.path.exists(config_path_yml) or os.path.exists(config_path_yaml):
+                self.directory = base_dir
+
+        # Figure out the repository type from the .nf-core.yml config file if we can
+        tools_config = nf_core.utils.load_tools_config(self.directory)
+        if tools_config.get("repository_type") in ["pipeline", "modules"]:
+            if self.repo_type is not None and self.repo_type != tools_config["repository_type"]:
+                raise UserWarning(
+                    f"'--repo-type {self.repo_type}' conflicts with [i][red]repository_type[/]: [blue]pipeline[/][/] in '{os.path.relpath(config_path_yml)}'"
+                )
+            self.repo_type = tools_config["repository_type"]
```

Comment:
I don't completely understand the question.. 🤔 How do you mean `Warning`?

The `raise UserWarning` gives a `CRITICAL` log message and exits (the `UserWarning` exception type is caught in upstream code to exit nicely without a traceback).

If we don't go into the `if` block here (no conflict) then we set `self.repo_type` and exit the function, all is well.

Reply:
The behavior is perfect then. 

For me the semantics of `Warning` is that it doesn't terminate the program, but just notifies the user, e.g. via `warnings.warn`, that's why I was confused at first.

Reply:
Yeah, this is just an arbitrary python exception type that I had previously picked. I never really know what builtin is best for these! I guess it might be best to create a custom set of exceptions and then use those.. 🤔